src/rgw/rgw_op.cc

```diff
@@ -4451,6 +4477,19 @@ void RGWPutObj::execute(optional_yield y)
     }
   }
 
+  RGWBucketSyncPolicyHandlerRef policy_handler;
+  op_ret = driver->get_sync_policy_handler(this, std::nullopt, s->bucket->get_key(), &policy_handler, s->yield);
+
+  if (op_ret < 0) {
+    ldpp_dout(this, 0) << "failed to read sync policy for bucket: " << s->bucket << dendl;
+    return;
+  }
+  if (policy_handler && policy_handler->bucket_exports_object(s->object->get_name(), *obj_tags)) {
```

Comment:
it looks like `obj_tags` can be null here. seeing crashes in qa:
```
2024-06-21T02:14:03.883 INFO:tasks.rgw.client.0.smithi175.stdout:/usr/include/c++/11/bits/unique_ptr.h:407: typename std::add_lvalue_reference<_Tp>::type std::unique_ptr<_Tp, _Dp>::operator*() const [with _Tp = RGWObjTags; _Dp = std::default_delete<RGWObjTags>; typename std::add_lvalue_reference<_Tp>::type = RGWObjTags&]: Assertion 'get() != pointer()' failed.
2024-06-21T02:14:03.884 INFO:tasks.rgw.client.0.smithi175.stdout:*** Caught signal (Aborted) **
2024-06-21T02:14:03.884 INFO:tasks.rgw.client.0.smithi175.stdout: in thread 7faf3dc40640 thread_name:io_context_pool
2024-06-21T02:14:03.884 INFO:tasks.rgw.client.0.smithi175.stdout: ceph version 19.0.0-4398-g348e6f9d (348e6f9d58b67fa11bf46aaeb2d3fd0b12b25a07) squid (dev)
2024-06-21T02:14:03.884 INFO:tasks.rgw.client.0.smithi175.stdout: 1: /lib64/libc.so.6(+0x3e6f0) [0x7faf9d83e6f0]
2024-06-21T02:14:03.885 INFO:tasks.rgw.client.0.smithi175.stdout: 2: /lib64/libc.so.6(+0x8b94c) [0x7faf9d88b94c]
2024-06-21T02:14:03.885 INFO:tasks.rgw.client.0.smithi175.stdout: 3: raise()
2024-06-21T02:14:03.885 INFO:tasks.rgw.client.0.smithi175.stdout: 4: abort()
2024-06-21T02:14:03.885 INFO:tasks.rgw.client.0.smithi175.stdout: 5: radosgw(+0x400cc8) [0x55ec7da07cc8]
2024-06-21T02:14:03.885 INFO:tasks.rgw.client.0.smithi175.stdout: 6: (RGWPutObj::execute(optional_yield)+0x4647) [0x55ec7dc8c177]
2024-06-21T02:14:03.885 INFO:tasks.rgw.client.0.smithi175.stdout: 7: (rgw_process_authenticated(RGWHandler_REST*, RGWOp*&, RGWRequest*, req_state*, optional_yield, rgw::sal::Driver*, bool)+0xaaa) [0x55ec7db1faaa]
2024-06-21T02:14:03.885 INFO:tasks.rgw.client.0.smithi175.stdout: 8: (process_request(RGWProcessEnv const&, RGWRequest*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, RGWRestfulIO*, optional_yield, rgw::dmclock::Scheduler*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >*, std::chrono::duration<unsigned long, std::ratio<1l, 1000000000l> >*, int*)+0xfed) [0x55ec7db212ad]
2024-06-21T02:14:03.885 INFO:tasks.rgw.client.0.smithi175.stdout: 9: radosgw(+0xe3a11f) [0x55ec7e44111f]
2024-06-21T02:14:03.885 INFO:tasks.rgw.client.0.smithi175.stdout: 10: radosgw(+0x477816) [0x55ec7da7e816]
2024-06-21T02:14:03.885 INFO:tasks.rgw.client.0.smithi175.stdout: 11: radosgw(+0x459dc4) [0x55ec7da60dc4]
2024-06-21T02:14:03.885 INFO:tasks.rgw.client.0.smithi175.stdout: 12: make_fcontext()
```

Reply:
Updated to use the safer `.get()` and edited related methods. Please re-test when possible